syntax = "proto2";

package lego.client.protobuf;

message GetVpnInfoRequest {
    optional bytes pubkey = 1;
}

message GetVpnInfoResponse {
	optional bytes ip = 1;
	optional uint32 port = 2;
	optional bytes encrypt_type = 3;
	optional bytes passwd = 4;  // encrypt with ecdh key
	optional bytes pubkey = 5;
}

message NewTx {
	optional bytes gid = 1;
	optional bytes from_acc_addr = 2;
	optional bytes from_pubkey = 3;
	optional bytes from_sign = 4;
	optional bytes to_acc_addr = 5;
	optional uint64 lego_count = 6;
}

message TxBft {
	optional NewTx new_tx = 1;
}

message BftMessage {
	optional bytes gid = 1;
	optional uint64 rand = 2;
	optional int32 status = 3;
	optional bool leader = 4;
	optional uint32 net_id = 5;
	optional bytes node_id = 6;
	optional bytes pubkey = 7;
	optional bytes sign_challenge = 8;
	optional bytes sign_response = 9;
	optional bytes secret = 10;
	optional bytes challenge = 11;
	optional bytes response = 12;
	optional bytes agg_sign_challenge = 13;
	optional bytes agg_sign_response = 14;
	repeated uint64 bitmap = 15;
	optional bool agree = 16;
	optional bytes bft_address = 17;
	optional uint32 pool_index = 18;
	optional bytes data = 19;
}

message ServiceMessage {
	optional GetVpnInfoRequest vpn_req = 1;
	optional GetVpnInfoResponse vpn_res = 2;
}